### DISCUSSION TITLE
Composer / Packagist - How should a composer.json look for changelog extraction?

### DISCUSSION THREAD

**Discussion by Schleuse** [renovatebot/renovate#32347] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab

### Please tell us more about your question or problem

Hello :wave: 

I'm currently having trouble getting changelogs for our privately hosted Composer packages. We are running our own internal instance of a Packagist repository using [Satis](https://github.com/composer/satis). When Renovate creates a pull request for a dependency update, it doesn't display the changelog attached to the GitLab Release in our private GitLab instance. It does however work for public packages from packagist.org.

I'm wondering how I can indicate to Renovate where to fetch package changelogs. I've tried adding the `homepage` and `support.source` fields in the composer.json of the packages, but this hasn't helped.

From reviewing the source code, it seems that the sourceUrl is fetched from the `source.url` field in the response from the Packagist repository. Is this sourceUrl used for changelog extraction?

How should a composer.json file for a package be configured to enable changelog extraction from a GitLab Release?

Here's a sample package response from our package registry:

<details><summary>Package information in Registry</summary>

```json
{
  "packages": {
      "vendor/package": {
          "5.11.0": {
              "name": "vendor/package",
              "version": "5.11.0",
              "version_normalized": "5.11.0.0",
              "source": {
                  "type": "git",
                  "url": "git@gitlab.vendor.com:vendor/package.git",
                  "reference": "cfdced2b7be93a106b7e367eb1259dbc2be2b90a"
              },
              "dist": {
                  "type": "tar",
                  "url": "https://satis.vendor.com/dist/vendor/package/vendor-package-5.11.0-09c492.tar",
                  "reference": "cfdced2b7be93a106b7e367eb1259dbc2be2b90a",
                  "shasum": "f85d24dcdfda2c83a0f3fd5d3e8cd0c41e352a48"
              },
              "require": {
                  "php": ">=8.0 <8.4"
              },
              "require-dev": {
              },
              "time": "2024-11-05T16:01:42+00:00",
              "type": "symfony-bundle",
              "archive": {
                  "exclude": [
                      "/*",
                      "/.*",
                      "!/src",
                      "!/config",
                      "!/composer.json"
                  ]
              },
              "scripts": {
              },
              "description": "Some private package",
              "homepage": "https://gitlab.vendor.com/vendor/package/",
              "support": {
                  "source": "https://gitlab.vendor.com/svendor/package/",
                  "issues": "https://gitlab.vendor.com/vendor/package/-/issues"
              }
          }
      }
  }
}
```

</details>

### Logs (if relevant)

<details><summary>Logs</summary>

```

```

</details>

**Comment by rarkins** [comment]
The first thing you need to identify is whether Renovate has been able to detect a `sourceUrl` for the relevant package.

After that you'd check whether Renovate is skipping changelog lookup because it can't identify what type of server it is (e.g. GitLab) or some other reason.
